libs/nest/config/README.md

```diff
@@ -0,0 +1,232 @@
+- Helpers for JSON encoded environment variables.
+- Validation, transformation and TypeScript types with Zod.
+
+This configuration system should replace NX's environment files as the primary way to configure Nest modules in our APIs. This is especially useful as we create bigger and deeper module trees.
+
+## Basic usage
+
+First you define a configuration and a load function.
+
+```tsx
+import { defineConfig } from '@island.is/nest/config'
+
+export const someModuleConfig = defineConfig({
+  name: 'SomeModule',
+  load: (env) => ({
+    url: env.required('SOME_MODULE_URL', 'http://localhost:3000'),
+    secret: env.required('SOME_MODULE_SECRET'),
+    ttl: env.optionalJSON('SOME_MODULE_TTL') ?? 60,
+  }),
+})
+```
+
+These configuration definitions should ideally mirror [config blocks in our Helm DSL](https://github.com/island-is/infrastructure/issues/918).
+
+The `env` object has helpers to read values from the environment:
+
+```tsx
+interface EnvLoader {
+  required(envVariable: string, devFallback?: string): string
+  requiredJSON(envVariable: string, devFallback?: any): any
+  optional(envVariable: string): string | undefined
+  optionalJSON(envVariable: string): any | undefined
+}
+```
+
+These helpers have the following behaviour:
+
+- The `optional` helpers return data straight from environment variables, or undefined if some variable is not set. This is useful for optional configuration which you might have good defaults for.
+- The `required` helpers are designed for everything that needs to be configured in production. If something is missing, its behaviour depends on NODE_ENV:
+  - In production, it crashes the process after logging something like this:
+    `SomeModule is not configured. Missing environment variables: SOME_MODULE_URL, SOME_MODULE_SECRET`
+  - In development, it returns the `devFallback` argument. If there is no `devFallback` specified, the process keeps running with a partial configuration and `isConfigured === false`. It also logs the above message as a warning.
+- The `*JSON` variants try to parse the environment variable as JSON. This is very handy for numerical environment variables or more complex configuration. If you ever use these helpers, you should specify the expected types using the Zod schema. Any parsing issues are logged as error before crashing the process (with `requiredJSON`).
+
+### Injecting module config
+
+After defining the config, you can inject it into services using its key. By using the `ConfigType` interface you get a strongly typed configuration object.
+
+```tsx
+import { Inject } from '@nestjs/common'
+import { ConfigType } from '@island.is/nest/config'
+import { someModuleConfig } from './someModule.config'
+
+export class SomeModuleService {
+  constructor(
+    @Inject(someModuleConfig.KEY)
+    private config: ConfigType<typeof someModuleConfig>,
+  ) {
+    config.isConfigured // boolean
+    config.data // {url: string, secret: string, ttl: any} | undefined
```

Comment:
is this from an older version? currently it wouldn't have data unless your config type had a data property, right?

Reply:
Good catch. Will fix!